src/Symfony/Component/TypeInfo/Type/CompositeTypeInterface.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Symfony\Component\TypeInfo\Type;
+
+use Symfony\Component\TypeInfo\Type;
+
+interface CompositeTypeInterface
+{
+    /**
+     * @return list<Type>
+     */
+    public function getTypes(): array;
```

Comment:
what is the benefit of this interface ? The various composite types we have right now have **different** method for what the type composing them mean. So I don't see any use case for that interface.

Reply:
It's Mathias'. I just didn't copy/paste it from his PR and forgot to commit his comments/credits (fixed). You're incorrect, both composite types actually have common methods. I added them to the interface to make it clearer, and improved the docblocks. Basically, composite types provide methods to query their subtypes/parts.